### ISSUE TITLE
Publish yaml releases with and without kubeadm bootstrapper

### ISSUE THREAD

**Issue by rsmitty** [kubernetes-sigs/cluster-api#1686]
**User Story**

As a capi user and bootstrap provider dev, I'd like to be able to easily deploy the capi bits without kubeadm when new releases are published.

**Detailed Description**

This was talked about in the CAPI meeting today, but it'd be nice to be able to easily consume the CAPI release without kudeadm in the mix if possible. It sounded like during the discussion, there may be some other work to do around the CAPI controller trying to automatically spin up the bootstrap controller and we may need flags to disable that behavior. But that aside, having a conditional or two yamls published in release assets or some easy way to just say "no, I want to deploy my own bootstrapper" would be greatly appreciated from the Talos side. Honestly, I'd be pretty happy if there was a good way to just keep the `cluster-api-components.yaml` and `bootstrap-components.yaml` like each project currently publishes, just all within the same repo now.

/kind feature

**Comment by ncdc** [comment]
@rsmitty would you be interested in working on this?

**Comment by timothysc** [comment]
@hardikdr - I'd suspect this work would be of interest to you give some of our conversations.

**Comment by rsmitty** [comment]
Sure, I'm interested in poking at this. I think it'll be really nice to have for me.

**Comment by hardikdr** [comment]
> @hardikdr - I'd suspect this work would be of interest to you given some of our conversations.

Yes, that'd be of interest for me, I will try to follow-up on it with @rsmitty. Thanks.

**Comment by ncdc** [comment]
@rsmitty hi, any updates on this?

**Comment by rsmitty** [comment]
not yet, hoping to get started on this either later this afternoon or early next week.

**Comment by rsmitty** [comment]
to circle back a bit, I'm hacking on this now. I've got the working branch here: https://github.com/rsmitty/cluster-api/tree/issue-1686

**Comment by rsmitty** [comment]
As we suspected, there were flags needed to disable the creation of the kubeadm components. I submitted a PR (should be mentioned above) to address that first part of the issue. Once that is in place, I've got some kustomize I'll PR for discussion as well.

**Comment by fabriziopandini** [comment]
if we should support swappable components, from a clusterctl perspective I would prefer to have 1 YAML “batteries included” and then 1 YAML for each part
- core-provider
- bootstrap-provider
- control plane provider (TBD if this should become a different category of provider)

The batteries included yaml will be mostly for end-users, while clusterctl is going to use the single components (or they swappable alternatives) to compose the management cluster according to the user's request (values of the CLI flags)
